common/include/pcl/common/impl/eigen.hpp

```diff
@@ -248,6 +249,43 @@ pcl::computeCorrespondingEigenVector (const Matrix& mat, const typename Matrix::
     eigenvector = vec3 / std::sqrt (len3);
 }
 
+namespace pcl {
+namespace detail{
+template <typename Vector, typename Scalar>
+struct EigenVector {
+  Vector vector;
+  Scalar length;
+};
+
+/**
+ * @brief returns the unit vector along the largest eigen value as well as the
+ *        length of the largest eigenvector
+ * @tparam Vector Requested result type, needs to be explicitly provided and has
+ *                to be implicitly constructible from ConstRowExpr
+ * @tparam Matrix deduced input type providing similar in API as Eigen::Matrix
+ */
+template <typename Vector, typename Matrix> static EigenVector<Vector, typename Matrix::Scalar>
+get_largest_3x3_eigenvector (const Matrix scaledMatrix)
```

Comment:
Maybe rename according to our standards, i.e. camel case for function names and snake case for variable names?

```suggestion
getLargest3x3Eigenvector(const Matrix scaled_matrix)
```